docs/core/docker/building-net-docker-images.md

```diff
@@ -2,224 +2,255 @@
+dotnet run
+```
 
-**Dockerfile.debug** - this file is based on the **microsoft/dotnet:1.0.0-preview2-sdk** image which if you note from the list of image variants, includes the SDK, CLI and .NET Core and will be the image used for development and debugging (F5). Including all of these components produces a larger image with a size roughly of 540MB.
+After the application starts, visit **http://localhost:5000** in your web browser.
 
-**Dockerfile** - this image is the release image based on **microsoft/dotnet:1.0.0-core** and should be used for production. This image when built is approximately 253 MB.
+### Build and run the sample with Docker for Linux containers
 
-### Creating the Docker images
-Using the `dockerTask.sh` or `dockerTask.ps1` script, we can build or compose the image and container for the **api** application for a specific environment. Build the **debug** image by running the following command.
+You can build and run the sample in Docker using Linux containers using the following commands (The instructions assume the root of the repository):
 
-```bash
-./dockerTask.sh build debug
+```console
+cd aspnetapp
+docker build -t aspnetapp .
+docker run -it --rm -p 5000:80 --name aspnetcore_sample aspnetapp
 ```
 
-The image will build the ASP.NET application, run `dotnet restore` ([see note](#dotnet-restore-note)), add the debugger to the image, set an `ENTRYPOINT` and finally copy the app to the image. The result is a Docker image named *api* with a `TAG` of *debug*.  See the images on the machine using `docker images`.
+> [!Note] The `docker run` '-p' argument maps port 5000 on your local machine to port 80 in the container (the port mapping form is `host:container`). For more information, see the [docker run](https://docs.docker.com/engine/reference/commandline/exec/) reference on command-line parameters.
 
-```bash
-docker images
+After the application starts, visit **http://localhost:5000** in your web browser.
+
+### Build and run the sample with Docker for Windows containers
 
-REPOSITORY          TAG                  IMAGE ID            CREATED             SIZE
-api                 debug                70e89fbc5dbe        a few seconds ago   779.8 MB
+You can build and run the sample in Docker using Windows containers using the following commands (The instructions assume the root of the repository):
+
+```console
+cd aspnetapp
+docker build -t aspnetapp .
+docker run -it --rm --name aspnetcore_sample aspnetapp
 ```
 
-Another way to generate the image and run the application within the Docker container is to open the application in Visual Studio Code and use the debugging tools. 
+> [!IMPORTANT]
+> You must navigate to the **container IP address** (as opposed to http://localhost) in your browser directly when using Windows containers. You can get the IP address of your container with the following steps:
 
-Select the debugging icon in the View Bar on the left side of Visual Studio Code.
+* Open up another command prompt.
```

Comment:
These steps seem overly complicated.  The simple approach would be
`docker inspect --format '{{ .NetworkSettings.Networks.nat.IPAddress }}' aspnetcore_sample`

Reply:
Upon further inspection, there are the same number of steps.  The documented approach uses concepts that are more familiar to Windows users who are new to Docker.  The approach I mentioned is what someone more familiar to Docker would use.  Given these docs are more focused towards Docker noobs.  The current documentation is the preferred approach.

Please leave the doc as you have it and ignore my `docker inspect` suggestion.

Reply:
Ok, thanks @MichaelSimons.  I really appreciate your thoughtful comments!